src/resource_pool_texture.cpp

```diff
@@ -9,94 +9,20 @@ namespace wr
 	{
 	}
 
-	//! Loads a texture
-	TextureHandle TexturePool::Load(std::string_view path, bool srgb, bool generate_mips)
+	TextureHandle TexturePool::LoadFromCompressedMemory(char* data, size_t width, size_t height, std::string& texture_extension, bool srgb, bool generate_mips)
 	{
-		std::optional<std::string_view> extension = util::GetFileExtension(path);
-
-		if (std::string_view ext_int = extension.value(); extension.has_value())
-		{
-			Texture* texture;
-
-			if (ext_int.find("png") != std::string_view::npos)
-			{
-				texture = LoadPNG(path, srgb, generate_mips);
-			}
-			else if (ext_int.find("dds") != std::string_view::npos)
-			{
-				texture = LoadDDS(path, srgb, generate_mips);
-			}
-			else if (ext_int.find("hdr") != std::string_view::npos)
-			{
-				texture = LoadHDR(path, srgb, generate_mips);
-			}
-			else
-			{
-				LOGC("Texture {} not loaded. Format not supported.", path);
-			}
-
-			LOG("[TEXTURE LOADED] {}", path);
-
-			uint64_t texture_id = m_id_factory.GetUnusedID();
-
-			TextureHandle texture_handle;
-			texture_handle.m_pool = this;
-			texture_handle.m_id = texture_id;
-
-			m_unstaged_textures.insert(std::make_pair(texture_id, texture));
-
-			return texture_handle;
-		}
-		else
-		{
-			LOGC("Invalid path used as argument. Path: {}", path);
-		}
-	}
-
-	TextureHandle TexturePool::LoadFromMemory(char * data, int width, int height, TextureType type, bool srgb, bool generate_mips)
-	{
-		Texture* texture;
-		switch (type) {
-		case TextureType::DDS:
-			texture = LoadDDSFromMemory(data, width, srgb, generate_mips);
-			break;
-		case TextureType::PNG:
-			texture = LoadPNGFromMemory(data, width, srgb, generate_mips);
-			break;
-		case TextureType::HDR:
-			texture = LoadHDRFromMemory(data, width, srgb, generate_mips);
-			break;
-		case TextureType::RAW:
-			texture = LoadRawFromMemory(data, width, height, srgb, generate_mips);
-		default:
-			LOGC("Texture {} not loaded. Format not supported.");
-		}
-
-		uint64_t texture_id = m_id_factory.GetUnusedID();
-
-		TextureHandle texture_handle;
-		texture_handle.m_pool = this;
-		texture_handle.m_id = texture_id;
-
-		m_unstaged_textures.insert(std::make_pair(texture_id, texture));
-
-		return texture_handle;
-	}
-
-	TextureHandle TexturePool::LoadFromMemory(char * data, int width, int height, std::string texture_extension, bool srgb, bool generate_mips)
-	{
-		std::for_each(texture_extension.begin(), texture_extension.end(), [](char & c) {
+		std::for_each(texture_extension.begin(), texture_extension.end(), [](char& c) {
 			c = ::tolower(c);
 		});
 
 		TextureType type;
-		if (texture_extension.size() == 0 || height > 1)
-		{
-			type = TextureType::RAW;
-		}
-		else if (texture_extension.compare("png") == 0)
+
+		if (texture_extension.compare("png") == 0
```

Comment:
Why use std::compare here? its slower here than a normal string comp and is less readable.

Reply:
👍